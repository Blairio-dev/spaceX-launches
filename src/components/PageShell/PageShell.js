import PropTypes from "prop-types";
import React from "react";
import styled from "@emotion/styled";
import { breakpoints, colours } from "../../assets/tokens";

const StyledContainer = styled("div")`
  background-color: ${colours.white};
  max-width: 1440px;
  padding-left: 40px;
  padding-top: 24px;
  width: 1440px;

  @media (max-width: ${breakpoints.sm}) {
    padding-left: 16px;
    padding-top: 16px;
  }
`;

const StyledWrapper = styled("div")`
  background-color: ${colours.primaryBlue};
  display: flex;
  justify-content: center;
  min-height: 100vh;
  width: 100vw;
`;

const PageShell = ({ children }) => (
  <StyledWrapper>
    <StyledContainer>{children}</StyledContainer>
  </StyledWrapper>
);

PageShell.propTypes = {
  /** Child page content. */
  children: PropTypes.node.isRequired,
};

export { PageShell };
