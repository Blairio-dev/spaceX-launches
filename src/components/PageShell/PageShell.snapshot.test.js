import React from "react";
import renderer from "react-test-renderer";
import { PageShell } from "./PageShell";

const DEFAULT_PROPS = {
  children: (
    <div>
      <span>Child content</span>
    </div>
  ),
};

describe("PageShell Snapshots", () => {
  it("Renders with default props", () => {
    const tree = renderer.create(<PageShell {...DEFAULT_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
