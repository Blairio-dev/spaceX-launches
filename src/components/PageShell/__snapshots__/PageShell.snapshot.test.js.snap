// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageShell Snapshots Renders with default props 1`] = `
<div
  className="css-fvwhtc"
>
  <div
    className="css-yf24bg"
  >
    <div>
      <span>
        Child content
      </span>
    </div>
  </div>
</div>
`;
